code/game/objects/structures/windoor_assembly.dm

```diff
@@ -44,27 +43,28 @@
 /obj/structure/windoor_assembly/Move()
 	var/turf/T = loc
 	. = ..()
-	setDir(ini_dir)
 	move_update_air(T)
 
 /obj/structure/windoor_assembly/update_icon_state()
 	icon_state = "[facing]_[secure ? "secure_" : ""]windoor_assembly[state]"
 
 /obj/structure/windoor_assembly/CanAllowThrough(atom/movable/mover, turf/target)
 	. = ..()
-	if(get_dir(loc, target) == dir) //Make sure looking at appropriate border
+	if(.)
 		return
-	if(istype(mover, /obj/structure/window))
-		var/obj/structure/window/W = mover
-		if(!valid_window_location(loc, W.ini_dir))
-			return FALSE
-	else if(istype(mover, /obj/structure/windoor_assembly))
-		var/obj/structure/windoor_assembly/W = mover
-		if(!valid_window_location(loc, W.ini_dir))
-			return FALSE
-	else if(istype(mover, /obj/machinery/door/window) && !valid_window_location(loc, mover.dir))
+
+	if(get_dir(loc, target) == dir)
 		return FALSE
```

Comment:
This seems like an inappropriate conversion, if we're facing the same direction as the mover shouldn't we let the parent call decide?

Reply:
Good catch, copypasta prevails. I've tweaked this to match the original flow of logic.